GAE/src/org/waterforpeople/mapping/app/web/SurveyRestServlet.java

```diff
@@ -390,6 +390,27 @@ private SurveyInstanceDto findSurveyInstance(Long surveyInstanceId) {
             }
         }
         return dtoList;
+        
+    }
+    
+    /**
+     * lists all questions for a given questionGroup
+     *
+     * @param groupId
+     * @return
+     */
+    private List<QuestionDto> listQuestions(Long groupId) {
+        return listQuestionsWithLevelNames(qDao.listQuestionsByQuestionGroup(groupId, false).values());
```

Comment:
When I looked at  this, I felt that the nesting intuitively would be the other way around i.e. 

```
listGroupQuestionsWithLevelName() {
 listQuestions();
addLevelNames();
}
```